src/utils/netutils.cpp

```diff
@@ -157,56 +164,69 @@ namespace
     }
 } // namespace
 
-char const* resolve_ipv4_address(const char* ip, in_addr & s4_sin_addr)
+AddrInfoPtrWithDel_t
+resolve_ip_address(const char *ip, int port, bool is_ipv6_target_conn, const char **error_result) noexcept
 {
-    if (!inet_aton(ip, &s4_sin_addr)) {
-        struct addrinfo * addr_info = nullptr;
-        int               result    = getaddrinfo(ip, nullptr, nullptr, &addr_info);
-        if (result) {
-            char const* error = (result == EAI_SYSTEM) ? strerror(errno) : gai_strerror(result);
-            LOG(LOG_ERR, "DNS resolution failed for %s with errno = %d (%s)",
-                ip, (result == EAI_SYSTEM) ? errno : result, error);
-            return error;
+    AddrInfoPtrWithDel_t addr_info_ptr;
+    addrinfo *addr_info = nullptr;
+    addrinfo hints;
+    char port_buf[16] { };
+
+    std::memset(&hints, 0, sizeof(hints));
+    hints.ai_family = (is_ipv6_target_conn) ? AF_INET6 : AF_INET;
+    hints.ai_socktype = SOCK_STREAM;
+    std::to_chars(port_buf, port_buf + sizeof(port_buf), port);
+    if (int res = ::getaddrinfo(ip, port_buf, &hints, &addr_info))
+    {
+        const char *error = (res == EAI_SYSTEM) ?
+            strerror(errno) : gai_strerror(res);
+
+        if (error_result)
+        {
+            *error_result = error;
         }
-        s4_sin_addr.s_addr = (reinterpret_cast<sockaddr_in *>(addr_info->ai_addr))->sin_addr.s_addr; /*NOLINT*/
-        freeaddrinfo(addr_info);
+        LOG(LOG_ERR,
+            "DNS resolution failed for %s with errno = %d (%s)",
+            ip,
+            (res == EAI_SYSTEM) ? errno : res, error);
+        return nullptr;
     }
-    return nullptr;
+    addr_info_ptr.reset(addr_info);
+    return addr_info_ptr;
```

Comment:
-> `return AddrInfoPtrWithDel_t(addr_info)`
and remove `return nullptr`.

Reply:
Yes